### ISSUE TITLE
Implement null semantics in compiler and interpreter

### ISSUE THREAD

**Issue by jlapacik** [influxdata/flux#1326]
The SPEC states:

```
Any non-boolean operator that operates on basic types, returns _null_ when at least one of its operands is _null_.
This can be explained intuitively with the following table and by thinking of a null value as an unknown value.

| Expression       | Evaluates To | Because                                                                             |
| ---------------- | ------------ | ----------------------------------------------------------------------------------- |
| _null_ + 5       | _null_       | Adding 5 to an unknown value is still unknown                                       |
| _null_ * 5       | _null_       | Multiplying an unknown value by 5 is still unknown                                  |
| _null_ == 5      | _null_       | We don't know if an unknown value is equal to 5                                     |
| _null_ < 5       | _null_       | We don't know if an unknown value is less than 5                                    |
| _null_ == _null_ | _null_       | We don't know if something unknown is equal to something else that is also unknown  |

In other words, operating on something unknown produces something that is still unknown.
The only place where this is not the case is in boolean logic.

Because boolean types are nullable, Flux implements ternary logic as a way of handling boolean operators with _null_ operands.
Again, by interpreting a _null_ operand as an unknown value, we have the following definitions:

* not _null_ = _null_
* _null_ or false = _null_
* _null_ or true = true
* _null_ or _null_ = _null_
* _null_ and false = false
* _null_ and true = _null_
* _null_ and _null_ = _null_

And finally, because records are represented using object types, attempting to access a column whose value is unknown or missing from a record will also return _null_.
```

This issue will be closed when the interpreter and the compiler both implement the above semantics.

**Comment by jsternberg** [comment]
I'm assigning this to myself to update the `values/binary.go` file with null support, but I think the type system is going to be its own separate issue.

This is also related to #1348.

**Comment by jlapacik** [comment]
Closed via #1348 .
